src/main/java/org/openstreetmap/atlas/checks/validation/intersections/EdgeCrossingEdgeCheck.java

```diff
@@ -178,6 +199,40 @@ public boolean validCheckForObject(final AtlasObject object)
         return Optional.of(newFlag);
     }
 
+    /**
+     * Check if {@link Edge} highway tag is vehicle or pedestrian navigable.
+     *
+     * @param edge
+     *            Atlas object.
+     * @param carNav
+     *            Car Navigable Highway Type.
+     * @param pedestrianNav
+     *            Pedestrian Navigable Highway Type.
+     * @return {@code true} if {@link Edge} is vehicle or pedestrian navigable.
+     */
+    private boolean getCrossingHighwayType(final Edge edge, final boolean carNav,
+            final boolean pedestrianNav)
+    {
+        if (carNav && pedestrianNav)
+        {
+            return HighwayTag.isCarNavigableHighway(edge)
+                    || HighwayTag.isPedestrianNavigableHighway(edge);
+        }
+        else if (carNav)
+        {
+            return HighwayTag.isCarNavigableHighway(edge);
+        }
+        else if (pedestrianNav)
+        {
+            return HighwayTag.isPedestrianNavigableHighway(edge);
+        }
+        else
+        {
+            return !HighwayTag.isCarNavigableHighway(edge)
+                    && !HighwayTag.isPedestrianNavigableHighway(edge);
+        }
```

Comment:
I think this can just be the following, as I don't think we need to look for non car or ped edges if neither filter is true. 

```suggestion
        return (!carNav || HighwayTag.isCarNavigableHighway(edge)) && (!PedestrianNav || HighwayTag.isPedestrianNavigableHighway(edge) && (CarNav || PedestrianNav);
```